lib/practice_exercises.rb

```diff
@@ -1,13 +1,42 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) - two loops are run, but they are not nested so it is O(2n) which drops the 2 and becomes O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  return true if string == ""
+
+  new_string = ""
+  length = string.length
+  valid_characters = /\w/
+  length.times do |i|
+    if string[i].match?(valid_characters)
+      new_string += string[i]
+    end
+  end
+
+  length = new_string.length
+  new_string = new_string.downcase
+  length.times do |i|
+    first = new_string[i]
+    last = new_string[length-1]
+    return false if first != last
+    length -= 1
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) - there is a nested loop so this is O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word.  Otherwise 👍